### ISSUE TITLE
linter: `react/no-find-dom-node` not triggering on TS files, only TSX

### ISSUE THREAD

**Issue by JasonKaz** [oxc-project/oxc#22468]
### What version of Oxlint are you using?

1.64.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["react"],
  "rules": {
    "react/no-find-dom-node": "error"
  }
}
```

### What happened?

The `react/no-find-dom-node` rule does not find a violation when the code is in .ts files, only in .tsx files. This differs from eslint where the rules could be configured to run against .ts files.

Seems like a change is needed to [should_run()](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/react/no_find_dom_node.rs#L84) to check against non-jsx files?

demo.tsx
```ts
import React from 'react';
import ReactDOM from 'react-dom';

// oxlint WILL flag this file — .tsx triggers react plugin rules
class Demo extends React.Component {
    foo() {
        ReactDOM.findDOMNode(this); // react/no-find-dom-node fires here
    }
}
```

demo.ts
```ts
import React from 'react';
import ReactDOM from 'react-dom';

// oxlint will NOT flag this file — react rules only apply to .tsx (JSX-capable) files
class Demo extends React.Component {
    foo() {
        ReactDOM.findDOMNode(this); // no oxlint violation here
    }
}
```
